src/sentry/search/events/datasets/discover.py

```diff
@@ -995,6 +1007,54 @@ def _resolve_apdex_function(self, args: Mapping[str, str], alias: str) -> Select
 
         return Function("apdex", function_args, alias)
 
+    def _resolve_web_vital_function(
+        self, args: Mapping[str, str | Column], alias: str
+    ) -> SelectType:
+        column = args["column"]
+        quality = args["quality"].lower()
+
+        if column.subscriptable != "measurements" or column.key not in VITAL_THRESHOLDS:
+            raise InvalidSearchQuery(f"count_web_vitals doesn't support {column.key}")
```

Comment:
Correct me if I'm wrong but if you pass something like `transaction.duration`, wouldn't that pass the `NumericColumn` check in the definition but `column.key` would be `None`. It won't throw but the error message would not be useful.

Reply:
:+1: Updated to check if its a measurement or not instead, since we won't be able to have the public alias at this point